### ISSUE TITLE
Directories in path -> service-worker 404s

### ISSUE THREAD

**Issue by framp** [tatethurston/next-service-worker#1]
Thanks a lot for the convenient library!

This line: https://github.com/tatethurston/next-service-worker/blob/main/src/index.ts#L144 makes it hard to use next-service-worker when having directories in your application's path.

Eg. Accessing https://site.com/lol/cat will throw:

```
Uncaught (in promise) TypeError: Failed to register a ServiceWorker for scope ('https://site.com/') with script ('https://site.com/lol/service-worker.js'): A bad HTTP response code (404) was received when fetching the script.
```

Changing to `/service-worker` is a quick fix but I'm not sure about the implications for its intended usage.
I think making the name configurable from options would offer a potential solution for this issue and give a bit of extra freedom to the end user. 

`autoRegister: false` is a good temporary solution, albeit forcing you to add service-worker code outside of next.config.js

**Comment by tatethurston** [comment]
Hey @framp, a few questions so I better understand:

1. Do you intend for the service worker to control everything under `https://site.com/` or only under `https://site.com/lol`?
2. Are you using [basePath](https://nextjs.org/docs/api-reference/next.config.js/basepath) in your next.config.js?
3. Is your next app only served under `https://site.com/lol/`?

**Comment by framp** [comment]
Hey Tate,

1. My service worker should control everything under https://site.com
2. Nope, I'm not, my next application is deployed on the root
3. My next app is served on https://site.com, https://site.com/lol and https://site.com/lol/cat

To elaborate more:
- When I'm visiting https://site.com it fetches https://site.com/service-worker.js and it works fine.
- When I'm visiting https://site.com/lol or https://site.com/lol/cat the application attempts to fetch a service worker from https://site.com/lol/service-worker.js or https://site.com/lol/cat/service-worker.js and fails with a 404

I'm not sure if it's expected behaviour or if you would consider it a bug - but it's trivially fixed by setting `autoRegister: false` and loading manually from the root
```
if (typeof navigator !== "undefined" && "serviceWorker" in navigator) {
  const wb = new Workbox("/service-worker.js", { // note the '/' in front of service-worker.js
    scope: "/",
  });
  wb.register();
}
```

if the name of the service-worker file were configurable, I could have potentially solved it with

```
//next.config.js
module.exports = withServiceWorker({
    serviceWorker: {
      autoRegister: true,
      filePath '/service-worker.js', // filePath can default to the current 'service-worker.js'
      ...
   }
 })
```
retaining the benefit of having my service-worker configuration all in `next.config.js`

**Comment by tatethurston** [comment]
Thanks for reporting this Federico. This is a bug -- the default worker path is relative when it should be served from the root (`/service-worker.js` instead of `service-worker.js`). I've fixed this and published v0.0.4.

v0.0.4 also makes the name configurable, but this should not be necessary for your use case now -- things should work without configuration with this setup.

Let me know if you run into any issues!
